crates/bevy_render/src/texture/image.rs

```diff
@@ -613,6 +615,23 @@ impl CompressedImageFormats {
     }
 }
 
+// Convert an [`Image`] to `winit::window::Icon`.
+impl TryInto<Icon> for Image {
+    type Error = anyhow::Error;
```

Comment:
A previous PR had a purpose-specific error type, which might be good to do here (not sure).

Reply:
Yeah I'd prefer to use a custom error type over anyhow.